stream-projects/mogul-menu/components/menu/extension-icon/extension-icon.tsx

```diff
@@ -1,36 +1,66 @@
 import {
-  classKebab,
-  getSrcByImageObj,
-  React,
-  Ripple,
-  useStyleSheet,
+    classKebab,
+    getSrcByImageObj, jumper,
+    React,
+    Ripple, useEffect, useState,
+    useStyleSheet,
 } from "../../../deps.ts";
 import { getHasNotification, useTabs } from "../../tabs/mod.ts";
-import { getMenuIconImageObj, useMenu } from "../mod.ts";
+import { getMenuIconImageObj, useMenu, getIsOpen } from "../mod.ts";
 import stylesheet from "./extension-icon.scss.js";
+import hamburgerIcon from "../../../assets/hamburgerIcon.png";
 
 export default function ExtensionIcon() {
-  const { state: menuState, toggleOpen } = useMenu();
-  const { state: tabsState } = useTabs();
-  const hasNotification = getHasNotification(tabsState);
+    const { state: menuState, toggleOpen } = useMenu();
+    const { state: tabsState } = useTabs();
 
-  const onExtensionIconClick = () => {
+    const hasNotification = getHasNotification(tabsState);
+    const menuOpen = getIsOpen(menuState);
+
+    const onExtensionIconClick = () => {
     toggleOpen();
-  };
+    };
+
+    useStyleSheet(stylesheet);
+    const iconImageObj = getMenuIconImageObj(menuState);
+
+    const menuOpenStyles = {
+        justifyContent: 'space-between',
+        alignItems: 'left',
+        paddingLeft: 0,
+        width: 48,
+        borderWidth: 0
+    }
 
-  useStyleSheet(stylesheet);
-  const iconImageObj = getMenuIconImageObj(menuState);
-  return (
-    <div
-      className={`c-extension-icon ${classKebab({ hasNotification })}`}
-      style={{
-        backgroundImage: iconImageObj
-          ? `url(${getSrcByImageObj(iconImageObj)})`
-          : undefined,
-      }}
-      onClick={onExtensionIconClick}
-    >
-      <Ripple color="var(--mm-color-text-bg-primary)" />
-    </div>
-  );
+    return (
+      <div
+          className={`c-extension-icon ${classKebab({ hasNotification })}`}
+          style={{...(menuOpen && menuOpenStyles)}}
```

Comment:
is this the same as `style={menuOpen && menuOpenStyles}`? or does react freak out if style={false}

Reply:
It freaks out if I do that